server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/NewDuplicationMeasuresStep.java

```diff
@@ -0,0 +1,215 @@
+      Component leaf = context.getLeaf();
+      Iterable<Duplication> duplications = requireNonNull(this.duplicationRepository, "DuplicationRepository missing").getDuplications(leaf);
+      Optional<ScmInfo> scmInfo = scmInfoRepository.getScmInfo(leaf);
+
+      if (!scmInfo.isPresent()) {
+        return;
+      }
+
+      NewLinesDuplicatedAccumulator newLinesDuplicatedAccumulator = new NewLinesDuplicatedAccumulator(scmInfo.get(), context.getPeriods());
+
+      for (Duplication duplication : duplications) {
+        newLinesDuplicatedAccumulator.addBlock(duplication.getOriginal());
+        duplication.getDuplicates().stream()
+          .filter(InnerDuplicate.class::isInstance)
+          .map(duplicate -> (InnerDuplicate) duplicate)
+          .forEach(duplicate -> newLinesDuplicatedAccumulator.addBlock(duplicate.getTextBlock()));
+      }
+
+      Map<Period, Integer> newLinesDuplicatedByPeriod = newLinesDuplicatedAccumulator.getNewLinesDuplicated();
+      context.getPeriods().forEach(period -> newLines.increment(period, newLinesDuplicatedByPeriod.getOrDefault(period, 0)));
+    }
+  }
+
+  private static class NewLinesDuplicatedAccumulator {
+    private final ScmInfo scmInfo;
+    private final List<Period> periods;
+    private final SetMultimap<Period, Integer> counts;
+
+    private NewLinesDuplicatedAccumulator(ScmInfo scmInfo, List<Period> periods) {
+      this.scmInfo = scmInfo;
+      this.periods = periods;
+      this.counts = LinkedHashMultimap.create(periods.size(), Iterables.size(scmInfo.getAllChangesets()));
+    }
+
+    void addBlock(TextBlock textBlock) {
+      IntStream.rangeClosed(textBlock.getStart(), textBlock.getEnd())
+        .forEach(line -> periods.stream()
+          .filter(period -> isLineInPeriod(line, period))
+          .forEach(period -> counts.put(period, line)));
+    }
+
+    Map<Period, Integer> getNewLinesDuplicated() {
+      return ImmutableMap.copyOf(counts.keySet().stream().collect(toMap(Function.identity(), period -> counts.get(period).size())));
+    }
+
+    private boolean isLineInPeriod(int lineNumber, Period period) {
+      return scmInfo.getChangesetForLine(lineNumber).getDate() > period.getSnapshotDate();
+    }
+  }
+
+  private static final class NewDuplicationFormula implements Formula<NewDuplicationCounter> {
+    private static final Formula VIEW_FORMULA = new VariationSumFormula(NEW_LINES_DUPLICATED_KEY, viewsRestrictedPeriods(), 0.0d);
+
+    private final DuplicationRepository duplicationRepository;
+    private final ScmInfoRepository scmInfoRepository;
+
+    private NewDuplicationFormula(ScmInfoRepository scmInfoRepository, @Nullable DuplicationRepository duplicationRepository) {
+      this.duplicationRepository = duplicationRepository;
+      this.scmInfoRepository = scmInfoRepository;
+    }
+
+    public static Formula<?> from(@Nullable ScmInfoRepository scmInfoRepository, @Nullable DuplicationRepository duplicationRepository) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1452)